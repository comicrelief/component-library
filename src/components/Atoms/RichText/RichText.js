import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

export const Inner = styled.div`
  text-align: ${props => props.align};
`;

const RichText = ({ align = 'left', markup = '', ...rest }) => (
  <Inner
    align={align}
    dangerouslySetInnerHTML={{ __html: markup }}
    {...rest}
  />
);

RichText.propTypes = {
  align: PropTypes.oneOf(['left', 'center', 'right']),
  markup: PropTypes.string
};

export default RichText;
