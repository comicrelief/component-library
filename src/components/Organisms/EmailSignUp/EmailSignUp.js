import React, { useState } from 'react';
import PropTypes from 'prop-types';

import Select from '../../Atoms/SelectField/SelectField';
import Text from '../../Atoms/Text/Text';
import {
  ESUWrapper,
  TopCopyWrapper,
  Form,
  PrivacyCopyWrapper,
  InputField,
  ButtonWrapper,
  Title
} from './EmailSignUp.style';
import { Button } from '../Membership/Membership.style';

const EmailSignUp = ({
  title,
  topCopy,
  isSuccess,
  isSchools,
  schoolsCopy,
  selectItems,
  successCopy,
  errorMsg,
  privacyCopy,
  subscribe,
  validate,
  buttonColor,
  backgroundColor,
  ...rest
}) => {
  const [value, setValue] = useState('');
  const [age, setAge] = useState('');
  const [next, setNext] = useState(false);

  const handleSubmit = e => {
    // Just stopping weird things happening for now when the user submits with the enter key
    // rather than the Submit button
    e.preventDefault();
  };

  const schoolsSelect = (
    <>
      <Select
        hideLabel
        label="Please choose an option"
        placeholder="Please choose an option"
        description="Please choose an option"
        errorMsg={errorMsg}
        options={selectItems}
        value={age}
        onChange={event => setAge(event.target.value)}
      />
      <Text tag="p" weight="400" size="m">
        {schoolsCopy}
      </Text>
    </>
  );

  const subscriptionSchoolsForm = (
    <Form onSubmit={e => handleSubmit(e)}>
      {!next && (
        <>
          <InputField
            aria-label="Email address"
            name="email"
            id="email"
            hasAria={false}
            showLabel={false}
            type="email"
            errorMsg={errorMsg}
            label="email"
            placeholder="example@youremail.com"
            value={value}
            onChange={event => setValue(event.target.value)}
          />
          <ButtonWrapper backgroundColor={backgroundColor}>
            <Button
              as="input"
              type="submit"
              data-test="subscribe-button-school"
              color={buttonColor}
              onClick={() => validate({ email: value }) && setNext(true)}
              value=" Subscribe"
            />
          </ButtonWrapper>
        </>
      )}
      {next && (
        <>
          {schoolsSelect}
          <ButtonWrapper backgroundColor={backgroundColor}>
            <Button
              as="input"
              type="submit"
              data-test="subscribe-button-school"
              color={buttonColor}
              onClick={() =>
                validate({ email: value, age }) &&
                subscribe({ email: `${value}`, age: `${age}` })
              }
              value=" Subscribe"
            />
          </ButtonWrapper>
        </>
      )}
    </Form>
  );

  const subscriptionForm = (
    <Form onSubmit={e => handleSubmit(e)}>
      <InputField
        aria-label="Email address"
        name="email"
        id="email"
        hasAria={false}
        showLabel={false}
        type="email"
        errorMsg={errorMsg}
        label="email"
        placeholder="example@youremail.com"
        value={value}
        onChange={event => setValue(event.target.value)}
      />
      <ButtonWrapper backgroundColor={backgroundColor}>
        <Button
          as="input"
          data-test="subscribe-button"
          type="submit"
          color={buttonColor}
          onClick={() => validate({ email: value }) && subscribe(value)}
          value=" Subscribe"
        />
      </ButtonWrapper>
    </Form>
  );

  const privacyContainer = (
    <PrivacyCopyWrapper>{privacyCopy}</PrivacyCopyWrapper>
  );

  return (
    <ESUWrapper backgroundColor={backgroundColor} {...rest}>
      <Title tag="h2" size="xxl" weight="400" family="Anton" uppercase>
        {title}
      </Title>
      {(!next || isSuccess) && (
        <TopCopyWrapper>{isSuccess ? successCopy : topCopy}</TopCopyWrapper>
      )}
      {!isSuccess && (isSchools ? subscriptionSchoolsForm : subscriptionForm)}
      {privacyContainer}
    </ESUWrapper>
  );
};

EmailSignUp.propTypes = {
  /** title */
  title: PropTypes.string.isRequired,
  /** top copy */
  topCopy: PropTypes.node.isRequired,
  /** displayed copy when subscription is successful */
  successCopy: PropTypes.node.isRequired,
  /** boolean if true display successCopy */
  isSuccess: PropTypes.bool.isRequired,
  /** boolean if true display schools dropdown */
  isSchools: PropTypes.bool,
  /** email error message */
  errorMsg: PropTypes.string.isRequired,
  /** privacy copy */
  privacyCopy: PropTypes.node.isRequired,
  /** privacy copy */
  schoolsCopy: PropTypes.node,
  selectItems: PropTypes.PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string.isRequired,
      displayValue: PropTypes.string.isRequired
    })
  ),
  /** subscription function */
  subscribe: PropTypes.func.isRequired,
  /** validation function */
  validate: PropTypes.func.isRequired,
  /** background color */
  backgroundColor: PropTypes.string,
  /** button color */
  buttonColor: PropTypes.string
};

EmailSignUp.defaultProps = {
  backgroundColor: 'deep_violet_dark',
  buttonColor: 'red',
  isSchools: false,
  schoolsCopy: '',
  selectItems: []
};

export default EmailSignUp;
