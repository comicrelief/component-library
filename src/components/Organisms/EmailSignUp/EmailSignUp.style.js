import styled from 'styled-components';

import spacing from '../../../theme/shared/spacing';
import TextInput from './_TextInput';
import Text from '../../Atoms/Text/Text';

const ESUWrapper = styled.div`
  display: flex;
  flex-direction: column;
  font-size: ${({ theme }) => theme.fontSize('s')};
  color: ${({ theme }) => theme.color('white')};
  background-color: ${({ theme, backgroundColour }) => theme.color(backgroundColour)};
  padding: ${spacing('m')};
`;

const TopCopyWrapper = styled.div`
  display: flex;
  margin-bottom: ${spacing('md')};
`;

const ButtonWrapper = styled.div`
  margin-top: ${spacing('md')};
  button {
    background-color: ${({ theme, buttonColour }) => theme.color(buttonColour)};
  }
`;

const PrivacyCopyWrapper = styled.div`
  display: flex;
  flex-direction: column;
  margin-top: ${spacing('md')};

  p {
    font-size: ${({ theme }) => theme.fontSize('s')};
    line-height: ${({ theme }) => theme.fontSize('xl')};
    a {
      font-size: ${({ theme }) => theme.fontSize('s')};
      color: ${({ theme }) => theme.color('white')};
    }
  }
`;

const FormInner = styled.div`
  display: flex;
  flex-direction: column;
  margin: ${spacing('md')} 0;
`;

const NameWrapper = styled.div`
  display: flex;
  flex-direction: column;
  gap: 0;

  @media ${({ theme }) => theme.allBreakpoints('L')} {
    justify-content: start;
    flex-direction: ${({ columnLayout }) => (columnLayout ? 'column' : 'row')};
    gap: ${({ columnLayout }) => (columnLayout ? 0 : spacing('md'))};
)};

  }
`;

const InputField = styled(TextInput)`
  width: 100%;
  margin-bottom: ${spacing('md')};

  & > span:first-child {
    color: ${({ theme }) => theme.color('white')};
  }

  @media ${({ theme }) => theme.allBreakpoints('L')} {
    max-width: 290px;
  }
`;

const Title = styled(Text)`
  margin-bottom: ${spacing('m')};
`;

export {
  ESUWrapper,
  TopCopyWrapper,
  PrivacyCopyWrapper,
  ButtonWrapper,
  FormInner,
  InputField,
  NameWrapper,
  Title
};
