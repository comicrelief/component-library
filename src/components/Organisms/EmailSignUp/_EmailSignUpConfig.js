import * as yup from 'yup';

const FIELDS = {
  FIRST_NAME: 'firstName',
  LAST_NAME: 'lastName',
  EMAIL: 'email',
  AGE: 'age'
};

const validationSchema = yup.object({
  [FIELDS.FIRST_NAME]: yup
    .string()
    .required('Please enter your first name')
    .matches(
      /^[A-Za-z][A-Za-z' -]*$/,
      "This field only accepts letters and ' - and must start with a letter"
    )
    .max(25, 'Your first name must be between 1 and 25 characters'),
  [FIELDS.LAST_NAME]: yup
    .string()
    .required('Please enter your last name')
    .matches(
      /^[A-Za-z][A-Za-z' -]*$/,
      "This field only accepts letters and ' - and must start with a letter"
    )
    .max(50, 'Your first name must be between 1 and 50 characters'),
  [FIELDS.EMAIL]: yup
    .string()
    .required('Please enter your email address')
    .email('Please enter a valid email address')
    .max(100, 'Your email address must be between 1 and 100 characters long')
});

export { FIELDS, validationSchema };
