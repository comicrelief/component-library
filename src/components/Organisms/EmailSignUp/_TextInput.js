import React from 'react';
import PropTypes from 'prop-types';
import Input from '../../Atoms/Input/Input';

// TODO: This is a copy paste of the RHF friendly `TextInput` from Marketing Prefs.
// Perhaps it would be worthwhile refactoring this into a new `Atom` as a seperate PR.
const TextInput = ({
  fieldName,
  label,
  optional,
  fieldType,
  formContext,
  ...rest
}) => {
  const { errors, register } = formContext;

  const props = {
    name: fieldName,
    type: fieldType,
    label,
    placeholder: label,
    errorMsg: errors && errors[fieldName] && errors[fieldName].message,
    optional,
    'aria-required': !optional,
    ...rest
  };

  return <Input {...props} {...register(fieldName)} />;
};

TextInput.defaultProps = {
  optional: null,
  fieldType: 'text',
  formContext: null
};

TextInput.propTypes = {
  fieldName: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  optional: PropTypes.bool,
  fieldType: PropTypes.string,
  formContext: PropTypes.shape()
};

export default TextInput;
