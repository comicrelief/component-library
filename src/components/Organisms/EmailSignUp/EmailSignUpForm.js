// EmailSignUpForm.js is just a wrapper for viewing in the CL
// The email sign up component that is used/exported is _EmailSignUp.js
import React from 'react';
import { useForm, FormProvider } from 'react-hook-form';
import { yupResolver } from '@hookform/resolvers/yup';
import RichText from '../../Atoms/RichText/RichText';
import {
  EmailSignUp,
  buildEsuValidationSchema,
  ESU_FIELDS
} from './_EmailSignUp';

const EmailSignUpForm = () => {
  const validationSchema = buildEsuValidationSchema({});
  const formMethods = useForm({
    mode: 'onBlur',
    resolver: yupResolver(validationSchema)
  });
  const { handleSubmit, trigger, setError } = formMethods;

  async function handleSubscribe(data) {
    const valid = await trigger([
      ESU_FIELDS.EMAIL,
      ESU_FIELDS.FIRST_NAME,
      ESU_FIELDS.LAST_NAME
    ]);
    if (valid) {
      // eslint-disable-next-line no-console
      console.log(data);
    } else {
      setError('formError', { message: 'Error' });
    }
  }
  const title = 'Stay in the know!';
  const topCopy = (
    <RichText
      markup={"<p>Get regular email updates and info on what we're up to!</p>"}
    />
  );
  const privacyCopy = (
    <RichText
      markup={
        '<p>Our <a class="link link--white inline" href="/privacy-notice">Privacy Policy</a> describes how we handle and protect your information.<br><br>If you are under 18, please make sure you have your parentsâ€™ permission before providing us with any personal details.</p>'
      }
    />
  );
  const successCopy = (
    <RichText markup="<p>Thanks! Your first email will be with you shortly</p>" />
  );
  return (
    <FormProvider {...formMethods}>
      <form onSubmit={handleSubmit(handleSubscribe)} noValidate>
        <EmailSignUp
          id="default"
          title={title}
          topCopy={topCopy}
          successCopy={successCopy}
          privacyCopy={privacyCopy}
          formContext={formMethods}
        />
      </form>
    </FormProvider>
  );
};
export default EmailSignUpForm;
