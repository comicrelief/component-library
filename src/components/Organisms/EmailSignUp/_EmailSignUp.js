import React from 'react';
import PropTypes from 'prop-types';
// import Select from '../../Atoms/Select/Select';
// import Text from '../../Atoms/Text/Text';
import {
  ESUWrapper,
  TopCopyWrapper,
  FormInner,
  PrivacyCopyWrapper,
  InputField,
  ButtonWrapper,
  Title
} from './EmailSignUp.style';
import { Button } from '../Membership/Membership.style';
import Text from '../../Atoms/Text/Text';
import { buildEsuValidationSchema, FIELDS } from './_EmailSignUpConfig';

const EmailSignUp = ({
  title,
  topCopy,
  isSchools,
  schoolsCopy,
  successCopy,
  privacyCopy,
  buttonColor,
  backgroundColor,
  formContext,
  ...rest
}) => (
  <ESUWrapper backgroundColor={backgroundColor} {...rest}>
    <Title tag="h2" size="xxl" weight="400" family="Anton" uppercase>
      {title}
    </Title>
    <TopCopyWrapper>
      <Text>{topCopy}</Text>
    </TopCopyWrapper>
    <FormInner>
      <InputField
        fieldName={FIELDS.FIRST_NAME}
        id="first-name"
        type="text"
        label="First Name"
        placeholder="Lenny"
        formContext={formContext}
      />
      <InputField
        fieldName={FIELDS.LAST_NAME}
        id="last-name"
        type="text"
        label="Last Name"
        placeholder="Henry"
        formContext={formContext}
      />
      <InputField
        fieldName={FIELDS.EMAIL}
        id="email"
        type="email"
        label="Email Address"
        placeholder="lenny@comicrelief.com"
        formContext={formContext}
      />
      <ButtonWrapper>
        <Button
          as="input"
          type="submit"
          data-test={isSchools ? 'subscribe-button-school' : 'subscribe-button'}
          color={buttonColor}
          value=" Subscribe"
        />
      </ButtonWrapper>
    </FormInner>
    <PrivacyCopyWrapper>
      <Text>{privacyCopy}</Text>
    </PrivacyCopyWrapper>
  </ESUWrapper>
);

EmailSignUp.propTypes = {
  title: PropTypes.string.isRequired,
  topCopy: PropTypes.node.isRequired,
  successCopy: PropTypes.node.isRequired,
  isSchools: PropTypes.bool,
  errorMsg: PropTypes.string.isRequired,
  privacyCopy: PropTypes.node.isRequired,
  schoolsCopy: PropTypes.node,
  backgroundColor: PropTypes.string,
  buttonColor: PropTypes.string,
  formContext: PropTypes.shape()
};

EmailSignUp.defaultProps = {
  backgroundColor: 'deep_violet_dark',
  buttonColor: 'red',
  isSchools: false,
  schoolsCopy: '',
  formContext: null
};

export { EmailSignUp, buildEsuValidationSchema, FIELDS };
