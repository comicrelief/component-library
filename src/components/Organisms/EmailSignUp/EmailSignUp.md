# Email SignUp Form

```js
// EmailSignUpForm.js is just a wrapper for viewing in the CL
// The email sign up component that is used/exported is _EmailSignUp.js
import EmailSignUpForm from './EmailSignUpForm';
import Text from '../../Atoms/Text/Text';

<>
    <Text tag="p">This EmailSignUpForm component exists purely to show the EmailSignUp component functioning within the Component Library; applications are to provide their own react-hook-form form and validation, based on these.</Text>

    <EmailSignUpForm />
</>
```
