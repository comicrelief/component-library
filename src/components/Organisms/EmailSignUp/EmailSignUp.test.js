import React from 'react';
import 'jest-styled-components';
import TestRenderer from 'react-test-renderer';
import renderWithTheme from '../../../hoc/shallowWithTheme';
import EmailSignUp from './EmailSignUp';
import RichText from '../../Atoms/RichText/RichText';

const { act } = TestRenderer;

it('renders correctly', () => {
  const top =
    '<h1> Top Copy</h1> <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</p>';
  const success =
    '<h1> Success Copy</h1> <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</p>';
  const privacy =
    'check <a href="https://www.comicrelief.com/privacy-notice">Privacy policy</a>';
  const tree = renderWithTheme(
    <>
      <EmailSignUp
        title="sign up letter"
        topCopy={<RichText markup={top} />}
        successCopy={<RichText markup={success} />}
        isSuccess={false}
        errorMsg=""
        buttonColor="teal"
        privacyCopy={<RichText markup={privacy} />}
        subscribe={() => 'Done'}
        validate={() => true}
      />
    </>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders ESU School correctly', () => {
  const top =
    '<h1> Top Copy</h1> <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</p>';
  const success =
    '<h1> Success Copy</h1> <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</p>';
  const privacy =
    'check <a href="https://www.comicrelief.com/privacy-notice">Privacy policy</a>';
  const selectItems = [
    { value: '', displayValue: '-- Select age group --' },
    { value: 'Option one', displayValue: 'The first option' },
    {
      value: 'Option two',
      displayValue: 'The second option'
    },
    { value: 'Option three', displayValue: 'The third option' },
    { value: 'Option four', displayValue: 'The fourth option' }
  ];
  const mockNext = jest.fn();
  const tree = renderWithTheme(
    <>
      <EmailSignUp
        title="sign up letter"
        topCopy={<RichText markup={top} />}
        successCopy={<RichText markup={success} />}
        schoolsCopy="Now please select your teaching group so you get the right updates."
        selectItems={selectItems}
        isSuccess={false}
        isSchools
        errorMsg=""
        buttonColor="teal"
        privacyCopy={<RichText markup={privacy} />}
        subscribe={mockNext}
        validate={() => true}
      />
    </>
  );
  const input = tree.root.findAllByType('input')[0];
  input.value = 'test@test.com';

  act(() => {
    /* fire events that update state */
    tree.root.findAllByType('input')[1].props.onClick();
  });

  const treeJson = tree.toJSON();
  expect(treeJson).toMatchSnapshot();
});
