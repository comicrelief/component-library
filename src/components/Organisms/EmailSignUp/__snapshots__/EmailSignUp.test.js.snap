// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ESU School correctly 1`] = `
.c1 {
  font-size: 3rem;
  line-height: 3rem;
  text-transform: uppercase;
  font-weight: 400;
  font-family: 'Anton',Impact,sans-serif;
  -webkit-letter-spacing: 0.03rem;
  -moz-letter-spacing: 0.03rem;
  -ms-letter-spacing: 0.03rem;
  letter-spacing: 0.03rem;
}

.c5 {
  font-size: 1rem;
  line-height: 1rem;
  text-transform: inherit;
  line-height: normal;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

.c8 {
  font-size: 1.25rem;
  line-height: 1.25rem;
  text-transform: inherit;
  font-weight: 400;
  line-height: normal;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: #5C5C5E;
}

.c6 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c7 {
  width: 100%;
  font-size: 1.25rem;
  display: block;
  box-sizing: border-box;
  padding: 0.8rem 1.5rem;
  padding-right: 4rem;
  margin: 0;
  position: relative;
  height: 48px;
  font-weight: 400;
  background: #F4F3F5 url(mock.asset) calc(100% - 1.5rem) 14px/20px 1.5rem no-repeat;
  border: 1px solid;
  border-color: #E1E2E3;
  box-shadow: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  color: #000000;
  border-radius: 0.5rem;
  cursor: pointer;
}

.c7:focus {
  border: 1px solid #666;
}

.c12 {
  text-align: left;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 1rem;
  color: #FFFFFF;
  background-color: #2C0230;
}

.c9 {
  margin-top: 1rem;
}

.c9 input {
  text-align: center;
  border-color: #2C0230;
  width: 100%;
  font-size: 1rem;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 1rem;
}

.c11 p {
  font-size: 1rem;
  line-height: 2rem;
}

.c11 p a {
  font-size: 1rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 1rem 0;
}

.c2 {
  margin-bottom: 1.5rem;
}

.c10 {
  width: 100%;
  color: #FFFFFF;
  font-size: 1rem;
  font-weight: bold;
  cursor: pointer;
  padding: 1rem 0.5rem;
  background: #4ECDBE;
  border: none;
  border-radius: 100px;
}

.c10:active,
.c10:focus,
.c10:hover {
  background-color: #961D35;
}

@media (min-width:740px) {

}

@media (min-width:740px) {
  .c7 {
    max-width: 290px;
  }
}

@media (min-width:740px) {
  .c9 input {
    font-size: 1.25rem;
    max-width: 180px;
  }
}

@media (min-width:1024px) {
  .c10 {
    width: auto;
    padding: 1rem 2rem;
    margin: 0 auto 2rem;
  }
}

<div
  className="c0"
>
  <h2
    className="c1 c2"
    color="inherit"
    size="xxl"
  >
    sign up letter
  </h2>
  <form
    className="c3"
    onSubmit={[Function]}
  >
    <label
      className="c4"
    >
      <span
        className="c5 c6"
        color="inherit"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Please choose an option",
          }
        }
        size="s"
      />
      <select
        className="c7"
        defaultValue=""
        onChange={[Function]}
        placeholder="Please choose an option"
        value=""
      >
        <option
          disabled={true}
          value=""
        >
          Please choose an option
        </option>
        <option
          value=""
        >
          -- Select age group --
        </option>
        <option
          value="Option one"
        >
          The first option
        </option>
        <option
          value="Option two"
        >
          The second option
        </option>
        <option
          value="Option three"
        >
          The third option
        </option>
        <option
          value="Option four"
        >
          The fourth option
        </option>
      </select>
      
    </label>
    <p
      className="c8"
      color="inherit"
      size="m"
    >
      Now please select your teaching group so you get the right updates.
    </p>
    <div
      className="c9"
    >
      <input
        className="c10"
        color="teal"
        data-test="subscribe-button-school"
        onClick={[Function]}
        type="submit"
        value=" Subscribe"
      />
    </div>
  </form>
  <div
    className="c11"
  >
    <div
      className="c12"
      dangerouslySetInnerHTML={
        Object {
          "__html": "check <a href=\\"https://www.comicrelief.com/privacy-notice\\">Privacy policy</a>",
        }
      }
    />
  </div>
</div>
`;

exports[`renders correctly 1`] = `
.c1 {
  font-size: 3rem;
  line-height: 3rem;
  text-transform: uppercase;
  font-weight: 400;
  font-family: 'Anton',Impact,sans-serif;
  -webkit-letter-spacing: 0.03rem;
  -moz-letter-spacing: 0.03rem;
  -ms-letter-spacing: 0.03rem;
  letter-spacing: 0.03rem;
}

.c8 {
  font-size: 1rem;
  line-height: 1rem;
  text-transform: inherit;
  line-height: normal;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: #5C5C5E;
}

.c9 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c11 {
  position: relative;
  box-sizing: border-box;
  width: 100%;
  height: 48px;
  padding: 1rem 1.5rem;
  background-color: #F4F3F5;
  border: 1px solid;
  border-color: #E1E2E3;
  box-shadow: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  color: #000000;
  border-radius: 0.5rem;
  font-size: inherit;
}

.c11:focus {
  border: 1px solid #666;
}

.c11:focus::-webkit-input-placeholder {
  color: #666;
}

.c11:focus::-moz-placeholder {
  color: #666;
}

.c11:focus:-ms-input-placeholder {
  color: #666;
}

.c11:focus::placeholder {
  color: #666;
}

.c10 {
  position: relative;
  font-size: 1.25rem;
}

.c4 {
  text-align: left;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 1rem;
  color: #FFFFFF;
  background-color: #2C0230;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1rem;
}

.c12 {
  margin-top: 1rem;
}

.c12 input {
  text-align: center;
  border-color: #2C0230;
  width: 100%;
  font-size: 1rem;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 1rem;
}

.c14 p {
  font-size: 1rem;
  line-height: 2rem;
}

.c14 p a {
  font-size: 1rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 1rem 0;
}

.c7 {
  width: 100%;
}

.c2 {
  margin-bottom: 1.5rem;
}

.c13 {
  width: 100%;
  color: #FFFFFF;
  font-size: 1rem;
  font-weight: bold;
  cursor: pointer;
  padding: 1rem 0.5rem;
  background: #4ECDBE;
  border: none;
  border-radius: 100px;
}

.c13:active,
.c13:focus,
.c13:hover {
  background-color: #961D35;
}

@media (min-width:740px) {
  .c11 {
    max-width: 290px;
  }
}

@media (min-width:740px) {
  .c12 input {
    font-size: 1.25rem;
    max-width: 180px;
  }
}

@media (min-width:1024px) {
  .c13 {
    width: auto;
    padding: 1rem 2rem;
    margin: 0 auto 2rem;
  }
}

<div
  className="c0"
>
  <h2
    className="c1 c2"
    color="inherit"
    size="xxl"
  >
    sign up letter
  </h2>
  <div
    className="c3"
  >
    <div
      className="c4"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h1> Top Copy</h1> <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</p>",
        }
      }
    />
  </div>
  <form
    className="c5"
    onSubmit={[Function]}
  >
    <label
      className="c6 c7"
      htmlFor="email"
    >
      <span
        className="c8 c9"
        color="inherit"
        dangerouslySetInnerHTML={
          Object {
            "__html": "email",
          }
        }
        size="s"
      />
      <div
        className="c10"
      >
        
        <input
          aria-label="Email address"
          className="c11"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="example@youremail.com"
          type="email"
          value=""
        />
        
      </div>
    </label>
    <div
      className="c12"
    >
      <input
        className="c13"
        color="teal"
        data-test="subscribe-button"
        onClick={[Function]}
        type="submit"
        value=" Subscribe"
      />
    </div>
  </form>
  <div
    className="c14"
  >
    <div
      className="c4"
      dangerouslySetInnerHTML={
        Object {
          "__html": "check <a href=\\"https://www.comicrelief.com/privacy-notice\\">Privacy policy</a>",
        }
      }
    />
  </div>
</div>
`;
